### DISCUSSION TITLE
ALE is installed but not doing anything in Vim

### DISCUSSION THREAD

**Discussion by PsySc0rpi0n** [w0rp/ale#4767] [Q&A - Ask for help with problems] [answered]
Hello.
I'm using Debian Bookworm:
`Linux  6.1.0-20-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.85-1 (2024-04-11) x86_64 GNU/Linux`

and Vim:

> $ vim --version
> VIM - Vi IMproved 9.0 (2022 Jun 28, compiled May 04 2023 10:24:44)
> Included patches: 1-1378, 1499
> Modified by team+vim@tracker.debian.org
> Compiled by team+vim@tracker.debian.org
> Huge version with GTK3 GUI.  Features included (+) or not (-):
> +acl               +file_in_path      +mouse_urxvt       -tag_any_white
> +arabic            +find_in_path      +mouse_xterm       +tcl
> +autocmd           +float             +multi_byte        +termguicolors
> +autochdir         +folding           +multi_lang        +terminal
> -autoservername    -footer            -mzscheme          +terminfo
> +balloon_eval      +fork()            +netbeans_intg     +termresponse
> +balloon_eval_term +gettext           +num64             +textobjects
> +browse            -hangul_input      +packages          +textprop
> ++builtin_terms    +iconv             +path_extra        +timers
> +byte_offset       +insert_expand     +perl              +title
> +channel           +ipv6              +persistent_undo   +toolbar
> +cindent           +job               +popupwin          +user_commands
> +clientserver      +jumplist          +postscript        +vartabs
> +clipboard         +keymap            +printer           +vertsplit
> +cmdline_compl     +lambda            +profile           +vim9script
> +cmdline_hist      +langmap           -python            +viminfo
> +cmdline_info      +libcall           +python3           +virtualedit
> +comments          +linebreak         +quickfix          +visual
> +conceal           +lispindent        +reltime           +visualextra
> +cryptv            +listcmds          +rightleft         +vreplace
> +cscope            +localmap          +ruby              +wildignore
> +cursorbind        +lua               +scrollbind        +wildmenu
> +cursorshape       +menu              +signs             +windows
> +dialog_con_gui    +mksession         +smartindent       +writebackup
> +diff              +modify_fname      +sodium            +X11
> +digraphs          +mouse             +sound             -xfontset
> +dnd               +mouseshape        +spell             +xim
> -ebcdic            +mouse_dec         +startuptime       -xpm
> +emacs_tags        +mouse_gpm         +statusline        +xsmp_interact
> +eval              -mouse_jsbterm     -sun_workshop      +xterm_clipboard
> +ex_extra          +mouse_netterm     +syntax            -xterm_save
> +extra_search      +mouse_sgr         +tag_binary        
> -farsi             -mouse_sysmouse    -tag_old_static    
>    system vimrc file: "/etc/vim/vimrc"
>      user vimrc file: "$HOME/.vimrc"
>  2nd user vimrc file: "~/.vim/vimrc"
>       user exrc file: "$HOME/.exrc"
>   system gvimrc file: "/etc/vim/gvimrc"
>     user gvimrc file: "$HOME/.gvimrc"
> 2nd user gvimrc file: "~/.vim/gvimrc"
>        defaults file: "$VIMRUNTIME/defaults.vim"
>     system menu file: "$VIMRUNTIME/menu.vim"
>   fall-back for $VIM: "/usr/share/vim"
> Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/x86_64-linux-gnu -I/usr/include/gio-unix-2.0 -I/usr/include/atk-1.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -pthread -Wdate-time -g -O2 -ffile-prefix-map=/build/vim-JA6Vy9/vim-9.0.1378=. -fstack-protector-strong -Wformat -Werror=format-security -DSYS_VIMRC_FILE=\"/etc/vim/vimrc\" -DSYS_GVIMRC_FILE=\"/etc/vim/gvimrc\" -D_REENTRANT -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 
> Linking: gcc -Wl,-E -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -o vim -lgtk-3 -lgdk-3 -lz -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lSM -lICE -lXt -lX11 -lXdmcp -lSM -lICE -lm -ltinfo -lselinux -lcanberra -lsodium -lacl -lattr -lgpm -L/usr/lib -llua5.2 -Wl,-E -fstack-protector-strong -L/usr/local/lib -L/usr/lib/x86_64-linux-gnu/perl/5.36/CORE -lperl -ldl -lm -lpthread -lcrypt -L/usr/lib/python3.11/config-3.11-x86_64-linux-gnu -lpython3.11 -ldl -lm -L/usr/lib/x86_64-linux-gnu -ltcl8.6 -ldl -lz -lpthread -lm -lruby-3.1 -lm -L/usr/lib

I installed ALE using VimPlug and my Vim config file is in ~/.vimrc.
The contents of Vim config file are:

> $ cat .vimrc
> call plug#begin('\~/.vim/plugins')
> " The default plugin directory will be as follows:
> "   - Vim (Linux/macOS): '\~/.vim/plugged'
> "   - Vim (Windows): '\~/vimfiles/plugged'
> "   - Neovim (Linux/macOS/Windows): stdpath('data') . '/plugged'
> " You can specify a custom plugin directory by passing it as the argument
> "   - e.g. `call plug#begin('\~/.vim/plugged')`
> "   - Avoid using standard Vimdirectory names like 'plugin'
> 
> " Make sure you use single quotes
> " Linter plugin Dense-Analysis ALE
> Plug 'dense-analysis/ale'
> 
> " Light Line plguin itchyny LightLine
> Plug 'itchyny/lightline.vim'
> 
> " Auto Pairs plugin auto-pairs
> Plug 'LunarWatcher/auto-pairs'
> 
> 
> " Initialize plugin system
> " - Automatically executes `filetype plugin indent on` and `syntax enable`.
> call plug#end()
> 
> set number
> set shiftwidth=4
> set expandtab
> set tabstop=4
> set laststatus=2
> let g:lightline = {
>     \'colorscheme': 'solarized',
> \}
> "hi CursorLine cterm=NONE ctermbg=darkred ctermfg=white guibg=darkred guifg=white
> "hi CursorColumn cterm=NONE ctermbg=darkgreen ctermfg=white guibg=darkgreen guifg=white

When I open a file, let's say with Python code and make a mistake on purpose to see if linter is working, nothing happens.
![image](https://github.com/dense-analysis/ale/assets/5173990/bb3d27e9-2301-435d-aff2-969d95121f63)

What can I do to try to figure out what is wrong?

**Comment by hsanson** [comment, marked as answer]
Run `ALEInfo` that may provide some insights of what may be the problem. Also ensure you have installed and configured the linter you want to use.  See list of supported tools in the help documentation `:h ale-supported-list`.

**Comment by PsySc0rpi0n** [comment]
> Run `ALEInfo` that may provide some insights of what may be the problem. Also ensure you have installed and configured the linter you want to use. See list of supported tools in the help documentation `:h ale-supported-list`.

This is what `ALEInfo` shows:
```
 Current Filetype: 
Available Linters: []
  Enabled Linters: []
  Ignored Linters: []
 Suggested Fixers:
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 
 Global Variables:
" Press Space to read :help for a setting
let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = v:null
let g:ale_command_wrapper = v:null
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_disable_lsp = 'auto'
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = v:null
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_info_default_mode = 'preview'
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = v:null
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {}
let g:ale_linters_explicit = 0
let g:ale_linters_ignore = {}
let g:ale_list_vertical = v:null
let g:ale_list_window_size = v:null
let g:ale_loclist_msg_format = v:null
let g:ale_max_buffer_history_size = v:null
let g:ale_max_signs = v:null
let g:ale_maximum_file_size = v:null
let g:ale_open_list = v:null
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_root = {}
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = v:null
let g:ale_sign_error = v:null
let g:ale_sign_info = v:null
let g:ale_sign_offset = v:null
let g:ale_sign_style_error = v:null
let g:ale_sign_style_warning = v:null
let g:ale_sign_warning = v:null
let g:ale_sign_highlight_linenrs = v:null
let g:ale_type_map = v:null
let g:ale_use_neovim_diagnostics_api = 0
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 'all'
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
 
  Command History:
  ```

Maybe I don't have all things needed installed. For instance, if I want only for Python, C and bash (if posible), what should I install?
And these linters are debian packages that we can install with APT or is something else that is installed with ALE and/or with Vim?
I don't see any example so that I can try to follow so I'm a bit lost!

Edited;
Nevermind, I got it working now! At least the error checking is working. Completion I don't know yet.

**Comment by userwzm** [comment]
see in .md can find git address
